qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -1074,6 +1075,6 @@ def _run_analysis(
                         "ydata": series_data.y,
                         "sigma": series_data.y_err,
                     }
-                analysis_result["raw_data"] = raw_data_dict
+                result_data["raw_data"] = raw_data_dict
 
-        return [analysis_result], figures
+        return [CurveAnalysisResultData(result_data)], figures
```

Comment:
Why `CurveAnalysisResultData` is generated again with nested `CurveAnalysisResultData`?

Reply:
Sorry, I thought it was a dictionary, I'll fix. Also need to fix in T1 and T2Ramsey, where it should be `CurveAnalysisResultData` instead of `AnalysisResultData`.